(fix/status-go) fix profile picture update/removal

kounkou opened the pull request:
### Description

This PR is linked to [13197](https://github.com/status-im/status-desktop/pull/13197) and contains the status-go necessary changes to make work 
- the changing of the profile picture and
- the first step for being able to reset the profile picture (see below for more details)

This PR also contains generated files as we are augmenting the protobuf with the flag to know if the images should be considered deleted.

This PR fixes [9947](https://github.com/status-im/status-desktop/issues/9947)

### Design decisions

In this PR I evaluated Possible solutions in order to mark images as deleted :

### **[Preferred - Implemented]** Modify the client to handle the delete operation

This design decision makes more sense after discussion with Patryk and is simpler than the 2 steps delete below as it tackle the inner lack of handling of the delete profile operation.

### Execute 2 steps delete operation

The strategy is to execute the Delete in 2 steps.
Step 1 get the information of the contactcode advertisement, and fills the payload with the images.
Step 2 add a indicator that this is a delete operation
Step 2 is to perform the clean up of the Image to be deleted in the Database

### Generate an empty identity image

Sending the contactcode requires a payload, and because during the delete we don't have images, sending the default image would help work around the lack of image. This requires generating a default image and agreeing on Frontend and backend.
This approach also requires saving the image, so database resource is needed.

### Create a new API for the delete Image

Currently the sending of concactcode is done under several constraints that include having a payload. So the idea would be to trigger a deleteIndentity image that doesn't have all the constraints of the sending of contact code.
This might take longer to execute, and also I am not sure why we should use another API here, as we are still in the context of the contactcode.

### Known Limitations

- This is the reset is not persisted yet, meaning that at reboot, the reset changes are lost

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (16)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 7c29d4ea | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4570/1/) | 2024-01-16 10:03:17 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240116-095905-7c29d4-pr4570.zip) |
| :heavy_check_mark: | 7c29d4ea | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4570/1/) | 2024-01-16 10:03:56 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240116-095900-7c29d4-pr4570.zip) |
| :heavy_check_mark: | 7c29d4ea | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4570/1/) | 2024-01-16 10:04:48 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240116-095900-7c29d4-pr4570.aar) |
| :heavy_multiplication_x: | 7c29d4ea | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/1/) | 2024-01-16 10:05:14 | ~6 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/1//consoleText) |
| :heavy_multiplication_x: | 7c29d4ea | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/2/) | 2024-01-17 07:57:28 | ~5 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b5797bb7 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4570/2/) | 2024-01-19 23:44:02 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240119-234223-b5797b-pr4570.zip) |
| :heavy_check_mark: | b5797bb7 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4570/2/) | 2024-01-19 23:44:35 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240119-234223-b5797b-pr4570.aar) |
| :heavy_check_mark: | b5797bb7 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4570/2/) | 2024-01-19 23:45:35 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240119-234223-b5797b-pr4570.zip) |
| :heavy_multiplication_x: | b5797bb7 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/3/) | 2024-01-20 00:04:24 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 75431904 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4570/3/) | 2024-01-20 00:59:27 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240120-005620-754319-pr4570.zip) |
| :heavy_check_mark: | 75431904 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4570/3/) | 2024-01-20 00:59:31 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240120-005620-754319-pr4570.zip) |
| :heavy_check_mark: | 75431904 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4570/3/) | 2024-01-20 01:02:15 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240120-005620-754319-pr4570.aar) |
| :heavy_multiplication_x: | 75431904 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/4/) | 2024-01-20 01:17:02 | ~20 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/4//consoleText) |
| :heavy_multiplication_x: | 75431904 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/5/) | 2024-01-21 02:24:47 | ~5 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/5//consoleText) |
| :heavy_multiplication_x: | 75431904 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/6/) | 2024-01-21 08:04:25 | ~19 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/6//consoleText) |
| :heavy_multiplication_x: | 75431904 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/7/) | 2024-01-22 08:51:57 | ~19 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/7//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | a07948ce | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4570/4/) | 2024-01-23 08:37:14 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240123-083450-a07948-pr4570.aar) |
| :heavy_check_mark: | a07948ce | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4570/4/) | 2024-01-23 08:38:14 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240123-083454-a07948-pr4570.zip) |
| :heavy_check_mark: | a07948ce | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4570/4/) | 2024-01-23 08:38:19 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240123-083454-a07948-pr4570.zip) |
| :heavy_multiplication_x: | a07948ce | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/8/) | 2024-01-23 09:13:18 | ~38 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | aea1d517 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4570/5/) | 2024-01-24 19:35:09 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240124-193250-aea1d5-pr4570.zip) |
| :heavy_check_mark: | aea1d517 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4570/5/) | 2024-01-24 19:35:55 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240124-193245-aea1d5-pr4570.zip) |
| :heavy_check_mark: | aea1d517 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4570/5/) | 2024-01-24 19:36:01 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240124-193245-aea1d5-pr4570.aar) |
| :heavy_check_mark: | aea1d517 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/9/) | 2024-01-24 20:09:17 | ~36 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4570/9//consoleText) |

osmaczko reviewed (commented):
As discussed offline, the issue lies on the receiver's side, where we don't clean up images if they arrive empty.

kounkou commented on multiaccounts/database.go:
```diff
@@ -450,9 +451,29 @@ func (db *Database) publishOnIdentityImageSubscriptions(change *IdentityImageSub
+	return nil
+}
+
+func (db *Database) DeleteIdentityImage(keyUID string) error {
```
I had a discussion with @osmaczko, (thank you! 🙏🏾) and here is the summary:

Instead of introducing a flag and transmitting the images, we can effectively manage the absence of images within the `ContactChatIdentity` in the `messenger_handler`. This approach allows the recipient of the contact code to be informed about the change (deletion of the profile) and update the view accordingly.

Regarding the current code:

At present, we do not address the absence of images because our handling is focused solely on image updates. Enhancing the code to accommodate missing images should effectively address the reported issue. Additionally, adopting this approach will streamline the code, eliminating the need for managing a flag as proposed in this pull request.

kounkou commented:
> As discussed offline, the issue lies on the receiver's side, where we don't clean up images if they arrive empty.

I have updated the PR with the new revision that fix all above mentioned issues. Here is a video showing all the expected functionalities


[profile updates.webm](https://github.com/status-im/status-go/assets/2589171/60a24da8-3ff2-4efa-8d6f-b52f845c4e05)

osmaczko commented on protocol/persistence.go:
```diff
@@ -755,6 +755,22 @@ func (db sqlitePersistence) SaveContactChatIdentity(contactID string, chatIdenti
 		_ = tx.Rollback()
 	}()
 
+	if len(chatIdentity.Images) == 0 {
```
I am wondering if this is a suitable approach. In the future, we might want to remove only images of a specific type. In this scenario, an image will be removed only if there are no other images present at all.

The solution could be:
- collect all image types that are left
- then `DELETE FROM chat_identity_contacts WHERE contact_id = ? AND image_type NOT IN (?, ?, ...)`, `tx.Exec(contactID, imageTypes)`

kounkou replied:
Hey, thanks for your comment @osmaczko. I'll be reviewing the code with this new perspective. Seems to me like an amazing comment on my unknown unknowns 🙏🏾

kounkou replied:
It turns out, the keys to the chatIdentity.Images is actually a string that represents the type of images. This type could be one of :

```
SmallDimName = "thumbnail"
LargeDimName = "large"
BannerIdentityName = "banner"
```

The idea is to delete the image matching only the image type above.

-----

I have updated the code to handle the removing of images of specific type. I agree that this udpate makes the code more robust for future possible improvements like stated above, for instance when deleting 1 of 3 images.
I have also renamed the confusing `ImageType` to `ImageFormat` and retested the changes again like in the video
I believe those changes should be enough to complete this status-go PR. Sending another revision.
Thanks again!

kounkou replied:
I have updated the code to handle the removing of images of specific type. I agree that this udpate makes the code more robust for future possible improvements like stated above, for instance when deleting 1 of 3 images.
I have also renamed the confusing `ImageType` to `ImageFormat`
I believe those changes should be enough to complete this status-go PR. Sending another revision.
Thanks!

osmaczko reviewed (approved):
:muscle:

kounkou commented:
Once I obtain second approval, I will regenerate the files and merge.

MishkaRogachev commented on appdatabase/migrations/bindata.go:
```diff
@@ -188,7 +188,7 @@ func _1640111208_dummyUpSql() (*asset, error) {
 		return nil, err
 	}
 
-	info := bindataFileInfo{name: "1640111208_dummy.up.sql", size: 258, mode: os.FileMode(0644), modTime: time.Unix(1692726719, 0)}
```
If there are no changes in migrations, bindatas in the corresponding directories can be excluded from the commit

MishkaRogachev reviewed (approved):
Great work!

kounkou committed:
(fix/status-go) fix profile picture update/removal

This PR fixes [9947](status-im/status-desktop#9947) and contains :

    - Commit to fix the changing of custom picture and having the change
      reflected on contact's side
    - Commit to fix the deleting of picture and having the change reflected
      on contact's side
    - Rename confusing `ImageType` to `ImageFormat`
